llvm/lib/MC/MCObjectStreamer.cpp

```diff
@@ -902,7 +902,10 @@ void MCObjectStreamer::emitFileDirective(StringRef Filename,
                                          StringRef TimeStamp,
                                          StringRef Description) {
   getAssembler().addFileName(Filename);
-  // TODO: add additional info to integrated assembler.
+  if (!CompilerVerion.empty())
+    getAssembler().setCompilerVersion(CompilerVerion.data());
```

Comment:
Prefer str(), data() isn't necessarily null terminated according to the docs.

nit: can we fix the typo in the parameter name here while were at it? It shows up in the docs  🙂

```suggestion
    getAssembler().setCompilerVersion(CompilerVersion.str());
```

Reply:
Typo fixed in https://github.com/llvm/llvm-project/pull/80703